core/controllers/admin.py

```diff
@@ -127,6 +138,30 @@ def get(self):
     @require_super_admin
     def post(self):
         """Handles POST requests."""
+        def check_and_update_config_role(old_list, new_list, role):
+            """This function is used to update the role in user settings when
+            a role is changed using config domain. If a user is removed from
+            old_list, its role is changed to exploration editor. If a new user
+            is added to new_list, its role is changed to role
```

Comment:
also, the sentence is kind of confusing because "role" means two different things. Maybe "If a new user is added to new_list, its role is changed to the specified role." or something